### PR TITLE
fix package for the upcoming 1.7 release

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: src/base-traits.jl [hunk 1]
```diff
@@ -15,12 +15,16 @@ export IsLeafType, IsConcrete, IsBits, IsImmutable, IsContiguous, IsIndexLinear,
...
 @traitdef IsBits{X}
-Base.@pure _isbits(X) = X.isbitstype
+Base.@pure _isbits(X) = isbitstype(X)
 @traitimpl IsBits{X} <- _isbits(X)
 
 "Trait of all immutable types"
 @traitdef IsImmutable{X}
-Base.@pure _isimmutable(X) = !X.mutable
+if VERSION >= v"1.7.0-DEV.1279"
+    Base.@pure _isimmutable(x) = !(x.name.flags & 0x2 == 0x2)
```
Comment (mauro3) [inline src/base-traits.jl:24]: I don't follow.  What is this about?  Tnx!

Comment (KristofferC) [reply src/base-traits.jl:24]: https://github.com/JuliaLang/julia/pull/41018

But actually, I think just using `ismutabletype` works.

Comment (KristofferC) [reply src/base-traits.jl]: YOLO github edit...

File: src/base-traits.jl [hunk 2]
```diff
@@ -21,7 +21,7 @@ Base.@pure _isbits(X) = isbitstype(X)
 "Trait of all immutable types"
 @traitdef IsImmutable{X}
 if VERSION >= v"1.7.0-DEV.1279"
-    Base.@pure _isimmutable(x) = !(x.name.flags & 0x2 == 0x2)
+    Base.@pure _isimmutable(X) = !(Base.ismutabletype(X))
```
Comment (mauro3) [inline src/base-traits.jl:24]: Cool, that looks better!

File: src/base-traits.jl [hunk 3]
```diff
@@ -15,12 +15,16 @@ export IsLeafType, IsConcrete, IsBits, IsImmutable, IsContiguous, IsIndexLinear,
...
+Base.@pure _isbits(X) = isbitstype(X)
 @traitimpl IsBits{X} <- _isbits(X)
 
 "Trait of all immutable types"
 @traitdef IsImmutable{X}
-Base.@pure _isimmutable(X) = !X.mutable
+if VERSION >= v"1.7.0-DEV.1279"
+    Base.@pure _isimmutable(X) = !(Base.ismutabletype(X))
+else
+    Base.@pure _isimmutable(X) = isimmutable(X)
```
Comment (mauro3) [inline src/base-traits.jl:26]: Here the `!X.mutable` needs to stay as `isimmutable` works on the value and not the type.

Comment (KristofferC) [reply src/base-traits.jl]: yeah, not sure why I changed taht.

**Comment by KristofferC** [comment]
You need to approve the workflow to run I think.

**Comment by KristofferC** [comment]
👍

**Comment by mauro3** [comment]
Thanks Kristoffer!
Do you know off by hand what I need to add to Project.toml that on Julia1.7 older version do not get installed?

**Pull request merged**